scripts/run_commands.sh

```diff
@@ -0,0 +1,101 @@
+#
+# This scripts runs on the ec2 instance. It is uploaded to the EC2 instance by
+# test_commands.sh
+# The necessary env variables will be set by test_commands.sh using the
+# command line arguments provided to it.
+#
+
+declare -i FAILURES
+FAILURES=0
+
+RED='\033[1;31m'
+GREEN='\033[1;32m'
+LRED='\033[0;31m'
+LGREEN='\033[0;32m'
+NC='\033[0m' # No Color
+
+expect_success() {
+	if "${@}" &>> ~/ecs-cli-test-results/test_output.txt; then
+		echo -e "${GREEN}SUCCEEDED${NC}: ${LGREEN}${@}${NC}"
+	else
+		echo "-------"
+		echo -e "${RED}FAILED${NC}: ${LRED}${@}${NC}"
+		echo "-------"
+		FAILURES+=1
+	fi
+}
+
+# used for commands that whose arguments might contain sensitive information
+expect_success_no_log() {
+	if "${@}" &>> ~/ecs-cli-test-results/test_output.txt; then
+		echo -e "${GREEN}SUCCEEDED${NC}: ${LGREEN}${1} ${2}${NC}"
+	else
+		echo "-------"
+		echo -e "${RED}FAILED${NC}: ${LRED}${1} ${2}${NC}"
+		echo "-------"
+		FAILURES+=1
+	fi
+}
+
+rm ~/ecs-cli-test-results/test_output.txt # clean up from past tests (in case this instance has been used before)
+# create the results directory and file; in case this instance is being used for the first time
+mkdir ~/ecs-cli-test-results/
+touch ~/ecs-cli-test-results/test_output.txt
+
+if ! [ -z "${gitname}" ]; then
+	# Not testing local changes, testing a branch instead
+	echo "Testing $branch"
+	# install git and go
+	echo "TYPE y|yes then enter to proceed."
+	sudo yum install git go >> ~/ecs-cli-test-results/test_log.txt
+	# have to respond yes to prompt
+	# get CLI
+	export GOPATH="$HOME/go"
+	go get github.com/aws/amazon-ecs-cli >> ~/ecs-cli-test-results/test_log.txt
+	cd $GOPATH/src/github.com/aws/amazon-ecs-cli
+	url="https://github.com/"
+	url+=$gitname
+	url+="/amazon-ecs-cli.git"
+	git remote add fork $url &>> ~/ecs-cli-test-results/test_log.txt
+	git fetch fork &>> ~/ecs-cli-test-results/test_log.txt
+	git checkout "fork/${branch}"
+	make build
+
+	cd bin/local/
+fi
+
+# test commands
+
+# configure
+expect_success_no_log ./ecs-cli configure --region $region --access-key $access --secret-key $secret --cluster $cluster
+# up
+expect_success ./ecs-cli up --capability-iam --keypair $keypair --size 1 --instance-type t2.medium --force
+# create a service
+expect_success ./ecs-cli compose --file ./integration-tests/docker-compose.yml service up
+# take down service
+expect_success ./ecs-cli compose --file ./integration-tests/docker-compose.yml service down
+# create a task
+expect_success ./ecs-cli compose --file ./integration-tests/docker-compose.yml up
+# take down task
+expect_success ./ecs-cli compose --file ./integration-tests/docker-compose.yml down
+# take down cluster
+expect_success ./ecs-cli down --force
```

Comment:
This is a good start. We would want to add other tests such as `compose service ps`.

Reply:
Good point. I'll add tests for those. Any others that you think I should add for v1 of this script? (As time goes on I think we can add more complicated tests, but for now I want to make sure it at least tests all the basic functionality of the CLI).

Reply:
I think it would be good to have all the compose commands `start`,  `scale`, `create`, `run` and `stop` since they are commonly used and they all share some code, one Task Definition change might affect multiple commands. The other cluster related commands or ECR related commands can be done later.